packages/react/tooltip/src/Tooltip.tsx

```diff
@@ -241,12 +249,20 @@ const TooltipTrigger = React.forwardRef<TooltipTriggerElement, TooltipTriggerPro
           ref={composedTriggerRef}
           onMouseEnter={composeEventHandlers(props.onMouseEnter, context.onTriggerEnter)}
           onMouseLeave={composeEventHandlers(props.onMouseLeave, context.onTriggerLeave)}
-          onMouseDown={composeEventHandlers(props.onMouseDown, context.onTriggerLeave)}
-          onFocus={composeEventHandlers(props.onFocus, context.onTriggerFocus)}
-          onBlur={composeEventHandlers(props.onBlur, context.onTriggerLeave)}
+          onMouseDown={composeEventHandlers(props.onMouseDown, () => {
+            context.onClose();
+            isMouseDownRef.current = true;
+            document.addEventListener('mouseup', () => (isMouseDownRef.current = false), {
```

Comment:
I always get confused with those, what if the trigger unmounts between mousedown and mouseup? Then it's leaking no?

Reply:
Good spot. I've added a clean up for this!